api/envoy/config/wasm/v2/wasm.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.config.wasm.v2;
+
+option java_outer_classname = "WasmProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.config.wasm.v2";
+
+import "envoy/api/v2/core/base.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Wasm service]
+
+// Configuration for a Wasm VM.
+// [#next-free-field: 6]
+message VmConfig {
+  // An ID which will be used along with a hash of the wasm code (or null_vm_id) to determine which
+  // VM will be used for the plugin. All plugins which use the same vm_id and code will use the same
+  // VM. May be left blank.
+  string vm_id = 1;
+
+  // The Wasm runtime type (see source/extensions/common/wasm/well_known_names.h).
+  string runtime = 2;
```

Comment:
Shoud this be some extension alike setting? i.e.
```proto
message Runtime {
  string name = 1;
  google.protobuf.Any config = 2;
}
```

and then `configuration`, `allow_precompiled` below should be enclosed in to the typed `Any` config.

Reply:
All the runtimes that I am aware of support AOT compilation so this is a generic feature AFAICT.  However, for runtime specific features the proto.Any is an interesting idea.  What about "oneof" as an alternative?  Any is opaque which makes the configuration less explicit.  What sort of support (if any) is required to get Any working with yaml?

Reply:
We already do tens of Any supporting for the extensions like filters. You just need define message types which can be packed into Any, and use source/common/config/utility.h to convert (name, config) pare to your desired type. That will automatically handle TypedStruct etc as well.

Reply:
Done.